rpcs3/Emu/RSX/VK/VKTextureCache.h

```diff
@@ -890,8 +889,8 @@ namespace vk
 							range_reset = true;
 						}
 
-						// Upgrade to writer lock
-						lock.upgrade();
+						//Warning: Will deadlock if a double fault occurs and we enter here from 'flush' method
```

Comment:
Reader lock is not recursive and still can deadlock. You should make sure that the flush cannot trigger a page fault then.